From 7b9f6b6c9b984c32d9562e7f81696eb622f8d036 Mon Sep 17 00:00:00 2001
Message-Id: <7b9f6b6c9b984c32d9562e7f81696eb622f8d036.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Vidya Srinivas <vidya.srinivas@intel.com>
Date: Wed, 4 Jun 2014 08:43:09 +0530
Subject: [PATCH 086/312] FOR_UPSTREAM [VPG]: drm/i915: configure the display
 i2c controller.

The following changes have been ported:
	drm/i915: configure the display i2c controller
	(OriginalChangeID - Ief1d3dbeb687c8058e6ba9621121ae001bad96c7)
	Description: Add 100 retries and proper timeout for the i2c communication.

Issue: GMIN-1460
Change-Id: Id53d9e3af84d8f13c0b859652bed77d330485a5b
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
---
 drivers/gpu/drm/i915/intel_i2c.c |    3 +++
 drivers/i2c/algos/i2c-algo-bit.c |    5 ++++-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_i2c.c b/drivers/gpu/drm/i915/intel_i2c.c
index dd9f2dc..8c3dba4 100644
--- a/drivers/gpu/drm/i915/intel_i2c.c
+++ b/drivers/gpu/drm/i915/intel_i2c.c
@@ -595,6 +595,9 @@ int intel_setup_gmbus(struct drm_device *dev)
 
 		bus->adapter.owner = THIS_MODULE;
 		bus->adapter.class = I2C_CLASS_DDC;
+		/*TODO: Revisit and optimize this value */
+		bus->adapter.retries = 100;
+		bus->adapter.timeout = usecs_to_jiffies(2200);
 		snprintf(bus->adapter.name,
 			 sizeof(bus->adapter.name),
 			 "i915 gmbus %s",
diff --git a/drivers/i2c/algos/i2c-algo-bit.c b/drivers/i2c/algos/i2c-algo-bit.c
index 65ef966..46c1af59 100644
--- a/drivers/i2c/algos/i2c-algo-bit.c
+++ b/drivers/i2c/algos/i2c-algo-bit.c
@@ -530,8 +530,11 @@ static int bit_doAddress(struct i2c_adapter *i2c_adap, struct i2c_msg *msg)
 		if (flags & I2C_M_REV_DIR_ADDR)
 			addr ^= 1;
 		ret = try_address(i2c_adap, addr, retries);
-		if ((ret != 1) && !nak_ok)
+		if ((ret != 1) && !nak_ok) {
+			dev_err(&i2c_adap->dev,
+					"normal 7 bit addr failed\n");
 			return -ENXIO;
+		}
 	}
 
 	return 0;
-- 
1.7.9.5

